primitives/evm/src/lib.rs

```diff
@@ -57,17 +56,124 @@ pub struct Vicinity {
 	pub origin: H160,
 }
 
+/// `System::Account` 16(hash) + 20 (key) + 52 (AccountInfo::max_encoded_len)
+pub const ACCOUNT_BASIC_PROOF_SIZE: u64 = 88;
+/// `AccountCodesMetadata` read, temptatively 16 (hash) + 20 (key) + 40 (CodeMetadata).
+pub const ACCOUNT_CODES_METADATA_PROOF_SIZE: u64 = 76;
+/// 16 (hash1) + 20 (key1) + 16 (hash2) + 32 (key2) + 32 (value)
+pub const ACCOUNT_STORAGE_PROOF_SIZE: u64 = 116;
+/// Fixed trie 32 byte hash.
+pub const WRITE_PROOF_SIZE: u64 = 32;
+/// Account basic proof size + 5 bytes max of `decode_len` call.
+pub const IS_EMPTY_CHECK_PROOF_SIZE: u64 = 93;
+
+pub enum AccessedStorage {
+	AccountCodes(H160),
+	AccountStorages((H160, H256)),
+}
+
+#[derive(Clone, Copy, Eq, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Debug, Serialize, Deserialize))]
+pub struct WeightInfo {
+	pub ref_time_limit: Option<u64>,
+	pub proof_size_limit: Option<u64>,
+	pub ref_time_usage: Option<u64>,
+	pub proof_size_usage: Option<u64>,
```

Comment:
The `ref_time_limit` and `ref_time_usage` seems make things complicated. These two fields are only used in the primitives scope, as we don't need them in the evm/ethereum/dispatch-precompile scope, `weight_to_gas` takes care of that.  What the stack runner needs is a single one field `proof_size_usage` to record the proof_size extra cost, which is used to construct the ExecutionInfoV2.

Reply:
In the near future the plan is to benchmark ref_time cost for opcodes, and then record it as an external metric just like proof_size, and that's why they are there.